### PR TITLE
MRG: Support for non-FIFF files in Report.parse_folder

### PR DESCRIPTION
**Pull request by hoechenberger**
Support all non-FIFF files we can read when parsing a folder using `Report.parse_folder()`. Requested via https://mne.discourse.group/t/naming-convention-in-mne-vs-mne-bids-welcome-to-the-thunderdome/2467

### CODE REVIEW

**Review by agramfort — APPROVED** [review]
perfect !

@larsoner merge if you're happy

File: mne/tests/test_report.py [hunk 1]
```diff
@@ -170,6 +174,43 @@ def test_render_report(renderer, tmpdir):
...
+    report = Report()
+    report.parse_folder(data_path=tempdir, render_bem=False, on_error='raise')
+
+    # Check correct paths and filenames
+    for fname in fnames_out:
+        assert (op.basename(fname) in
+                [op.basename(x) for x in report.fnames])
+        assert (''.join(report.html).find(op.basename(fname)) != -1)
+
+    assert_equal(len(report.fnames), len(fnames_out))
```
Comment (larsoner) [inline mne/tests/test_report.py:200]: We prefer plain `assert` to `assert_equal` nowadays usually -- `assert_equal` is less informative than plain `assert len(report.fnames) == len(fnames_out)`. Feel free to remove all uses in this file if you want, but at least let's not add more of them

File: mne/tests/test_report.py [hunk 2]
```diff
@@ -170,6 +174,43 @@ def test_render_report(renderer, tmpdir):
...
+    assert_equal(len(report.html), len(report.fnames))
+    assert_equal(len(report.fnames), len(report))
+
+    report.data_path = tempdir
+    fname = op.join(tempdir, 'report.html')
+    report.save(fname=fname, open_browser=False)
+    with open(fname, 'rb') as fid:
+        html = fid.read().decode('utf-8')
```
Comment (larsoner) [inline mne/tests/test_report.py:208]: Why you do this and `''.join(report.html)` above? I would do one or the other once and use the result all places

File: doc/changes/latest.inc [hunk 3]
```diff
@@ -23,6 +23,8 @@ Enhancements
 
 - Speed up :func:`mne.inverse_sparse.tf_mixed_norm` using STFT/ISTFT linearity (:gh:`8697` by `Eric Larson`_)
 
+- `~mne.Report.parse_folder` now processes supported non-FIFF files by default, too (:gh:`8744` by `Richard Höchenberger`_)
```
Comment (larsoner) [inline doc/changes/latest.inc:26]: The `~` will make it show up just as `parse_folder` which will be less easy to read for non-MNE-experts than the full qualified name (or at least `Report.parse_folder`)
```suggestion
- `mne.Report.parse_folder` now processes supported non-FIFF files by default, too (:gh:`8744` by `Richard Höchenberger`_)
```

**Review by larsoner — COMMENTED** [review]
Just minor comments, otherwise LGTM!

**Comment by hoechenberger** [comment]
@larsoner I have addressed your comments.

File: mne/tests/test_report.py (hunk 1 above)
Comment (hoechenberger) [reply mne/tests/test_report.py]: I've replaced all calls to `assert_equal` with plain `assert` now

File: mne/tests/test_report.py (hunk 2 above)
Comment (hoechenberger) [reply mne/tests/test_report.py:206]: I've dropped the `join` line.

**Review by larsoner — APPROVED** [review]
LGTM feel free to merge when CIs come back happy, thanks @hoechenberger !

**Comment by agramfort** [comment]
maybe some CIs need to be restarted. Than let's merge

**Comment by hoechenberger** [comment]
One CI run failing again, this is caused by an upstream problem:
https://github.com/jupyter/nbformat/issues/206

**Pull request merged**